test/e2e/framework/pv_util.go

```diff
@@ -264,28 +318,49 @@ func CreatePVPVC(c clientset.Interface, pvConfig PersistentVolumeConfig, pvcConf
 	pvc := MakePersistentVolumeClaim(pvcConfig, ns)
 
 	// instantiate the pv
-	pv = createPV(c, pv)
+	if pv, err = createPV(c, pv); err != nil {
+		return nil, nil, err
+	}
 	// instantiate the pvc, handle pre-binding by VolumeName if needed
 	if preBind {
 		pvc.Spec.VolumeName = pv.Name
 	}
-	pvc = CreatePVC(c, ns, pvc)
-
-	return pv, pvc
+	if pvc, err = CreatePVC(c, ns, pvc); err != nil {
+		return pv, nil, err
+	}
+	return pv, pvc, nil
 }
 
 // Create the desired number of PVs and PVCs and return them in separate maps. If the
 // number of PVs != the number of PVCs then the min of those two counts is the number of
-// PVs expected to bind.
-func CreatePVsPVCs(numpvs, numpvcs int, c clientset.Interface, ns string, pvConfig PersistentVolumeConfig, pvcConfig PersistentVolumeClaimConfig) (PVMap, PVCMap) {
-
-	var i int
-	var pv *v1.PersistentVolume
-	var pvc *v1.PersistentVolumeClaim
+// PVs expected to bind. If a Create error occurs, all created PVs and PVCs are deleted
+// and the error is returned.
+func CreatePVsPVCs(numpvs, numpvcs int, c clientset.Interface, ns string, pvConfig PersistentVolumeConfig, pvcConfig PersistentVolumeClaimConfig) (PVMap, PVCMap, error) {
+
+	var (
+		i, extraPVs, extraPVCs int
+		pv                     *v1.PersistentVolume
+		pvc                    *v1.PersistentVolumeClaim
+		err                    error
+	)
 	pvMap := make(PVMap, numpvs)
 	pvcMap := make(PVCMap, numpvcs)
 
-	var extraPVs, extraPVCs int
+	handle_err := func() { // on error delete created resources, if any
```

Comment:
From @copejon 's comment, it looks like the callers are handling the partial cleanup in their AfterEach methods.  Either way, I would pick one and be consistent.  There are many other functions in this file that return objects from partially completed operations.